controllers/ocirepository_controller_test.go

```diff
@@ -240,265 +243,374 @@ func TestOCIRepository_Reconcile(t *testing.T) {
-	g.Expect(err).ToNot(HaveOccurred())
+	srv.StartTLS()
+	defer srv.Close()
 
-	repositoryURL := fmt.Sprintf("%s/podinfo", regServer.registryHost)
-	ociURL := fmt.Sprintf("oci://%s", repositoryURL)
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{},
+	}
+	// Use the server cert as a CA cert, so the client trusts the
+	// server cert. (Only works because the server uses the same
+	// cert in both roles).
+	pool := x509.NewCertPool()
+	pool.AddCert(srv.Certificate())
+	transport.TLSClientConfig.RootCAs = pool
+	transport.TLSClientConfig.Certificates = []tls.Certificate{clientTLSCert}
 
-	// Push Test Image
-	image = setPodinfoImageAnnotations(image, "6.1.6")
-	err = crane.Push(image, repositoryURL, crane.WithAuth(&authn.Basic{
-		Username: testRegistryUsername,
-		Password: testRegistryPassword,
-	}))
-	g.Expect(err).ToNot(HaveOccurred())
+	srv.Client().Transport = transport
+	pi2, err := createPodinfoImageFromTar("podinfo-6.1.5.tar", "6.1.5", srv.URL, []crane.Option{
+		crane.WithTransport(srv.Client().Transport),
+	}...)
+	g.Expect(err).NotTo(HaveOccurred())
 
-	// Test Image digest
-	podinfoImageDigest, err := image.Digest()
-	g.Expect(err).ToNot(HaveOccurred())
+	tlsSecretClientCert := corev1.Secret{
+		StringData: map[string]string{
+			CACert:     string(rootCertPEM),
+			ClientCert: string(clientCertPEM),
+			ClientKey:  string(clientKeyPEM),
+		},
+	}
 
 	tests := []struct {
 		name                  string
 		url                   string
+		tag                   string
 		digest                gcrv1.Hash
-		repoUsername          string
-		repoPassword          string
-		includeSecretRef      bool
-		includeServiceAccount bool
+		certSecret            *corev1.Secret
+		expectreadyconition   bool
+		expectedstatusmessage string
 	}{
 		{
-			name:                  "missing-auth",
-			url:                   ociURL,
-			repoUsername:          "",
-			repoPassword:          "",
-			digest:                podinfoImageDigest,
-			includeSecretRef:      false,
-			includeServiceAccount: false,
+			name:                  "test connection with CACert, Client Cert and Private Key",
+			url:                   pi2.url,
+			tag:                   pi2.tag,
+			digest:                pi2.digest,
+			certSecret:            &tlsSecretClientCert,
+			expectreadyconition:   true,
+			expectedstatusmessage: fmt.Sprintf("stored artifact for digest '%s'", pi2.digest.Hex),
 		},
 		{
-			name:                  "invalid-auth-via-secret",
-			url:                   ociURL,
-			repoUsername:          "InvalidUser",
-			repoPassword:          "InvalidPassword",
-			digest:                podinfoImageDigest,
-			includeSecretRef:      true,
-			includeServiceAccount: false,
+			name:                  "test connection with with no secret",
+			url:                   pi2.url,
+			tag:                   pi2.tag,
+			digest:                pi2.digest,
+			expectreadyconition:   false,
+			expectedstatusmessage: "failed to pull artifact",
```

Comment:
Is this case an intentional change compare to https://github.com/fluxcd/source-controller/blob/102e9a94e47b142fcbb475a829c138c31eadc0c7/controllers/ocirepository_controller_test.go#L658-L664 ?
Is it because of replacing the registry server with a common docker distribution server?

Reply:
`pi2` variable can be renamed as there's no `pi` in this test anymore.

Reply:
> Is it because of replacing the registry server with a common docker distribution server?

Nah, it still gives has this in the status message. I'll replace it.
The `failed to pull artifact` is the part of the message that we define so it is more predictable.